app/HutchAgent/HostedServices/QueuePollingHostedService.cs

```diff
@@ -36,6 +36,7 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
       using (var scope = _serviceProvider.CreateScope())
       {
         var queue = _serviceProvider.GetRequiredService<IQueueReader>();
+        var executeActionHandler = scope.ServiceProvider.GetService<ExecuteActionHandler>() ?? throw new InvalidOperationException();
```

Comment:
```suggestion
        var executeActionHandler = scope.ServiceProvider.GetRequiredService<ExecuteActionHandler>();
```

`GetRequiredService` throws for you instead of returning `null`, which is often desirable.

https://andrewlock.net/the-difference-between-getservice-and-getrquiredservice-in-asp-net-core/